src/network.cpp

```diff
@@ -38,6 +41,35 @@ inline String uptimeString() {
   return ret;
 }
 
+DynamicJsonDocument parseOwieStatusData() {
+  DynamicJsonDocument status(1024);
+
+  const uint16_t *cellMillivolts = relay->getCellMillivolts();
+  String out;
+  out.reserve(256);
+  for (int i = 0; i < 3; i++) {
+    out.concat("<tr>");
+    for (int j = 0; j < 5; j++) {
+      out.concat("<td>");
+      out.concat(cellMillivolts[i * 5 + j] / 1000.0);
+      out.concat("</td>");
+    }
+    out.concat("<tr>");
+  }
+
+  status["TOTAL_VOLTAGE"] =
+      String(relay->getTotalVoltageMillivolts() / 1000.0, 2) + "v";
+  status["CURRENT_AMPS"] = String(relay->getCurrentInAmps(), 1) + " Amps";
+  status["BMS_SOC"] = String(relay->getBmsReportedSOC()) + "%";
+  status["OVERRIDDEN_SOC"] = String(relay->getOverriddenSOC()) + "%";
+  status["USED_CHARGE_MAH"] = String(relay->getUsedChargeMah()) + " mAh";
+  status["REGENERATED_CHARGE_MAH"] =
+      String(relay->getRegeneratedChargeMah()) + " mAh";
+  status["UPTIME"] = uptimeString();
+  status["CELL_VOLTAGE_TABLE"] = out;
+  return status;
```

Comment:
A suggestion: In the spirit of better encapsulation, perform the serialization to string in this function instead of the call site as the caller doesn't require the JSON object.